### PR TITLE
Task/teva 1123 filter controls frontend for MAT dashboard

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-1123

## Changes in this PR:

- complete markup and add styling for filter controls
- add JS behaviour enhancements for filter controls
- make search filters functionality more flexible to use in this scenario

### CODE REVIEW

File: app/components/hiring_staff/no_vacancies_component.html.haml [hunk 1]
```diff
@@ -3,8 +3,6 @@
     .govuk-grid-column-full
       %h2.govuk-heading-m= t('schools.no_jobs.heading')
 
-      = link_to t('buttons.create_job'), new_organisation_job_path, class: 'govuk-button govuk-!-margin-bottom-0'
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/no_vacancies_component.html.haml:6]: I'm pretty sure this link still needs to be here

File: app/components/hiring_staff/vacancies_component.html.haml [hunk 2]
```diff
@@ -2,22 +2,16 @@
   .govuk-grid-column-one-quarter
     = form_for @filters_form, url: organisation_managed_organisations_path, html: { method: "patch" } do |f|
       = render 'shared/filter_group', f: f,
-        filters: { title: 'Job filters', items: [{ title: 'Locations', attribute: :managed_school_ids, options: @school_options, value_method: :id, text_method: :name, small: true }] }
+        filters: { title: 'Filter results', :removeButtons => true, :totalCount => @filters[:managed_school_ids]&.count, items: [{ title: 'Locations', key: 'locations', search: true, attribute: :managed_school_ids, selected: @filters[:managed_school_ids], options: @school_options, value_method: :id, text_method: :name, small: true }] }
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/vacancies_component.html.haml:5]: `remove_buttons: true, total_count: @filters[:managed_school_ids]&.count`

File: app/components/hiring_staff/vacancies_component.html.haml [hunk 3]
```diff
@@ -2,22 +2,16 @@
...
 
       = f.hidden_field :jobs_type, value: @selected_type
       = f.govuk_submit t('buttons.apply_filters')
 
 %div{ class: grid_column_class }
-  .dashboard-header
-    - if @organisation.is_a?(SchoolGroup)
-      .filters-information
-        %button.govuk-button.govuk-button--secondary{ class: "govuk-!-margin-bottom-0" }= t('buttons.hide_filters')
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/vacancies_component.html.haml:14]: Where has this button moved to?

File: app/frontend/src/lib/ui/components/filterGroup.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
+.moj-filter {
+
+  box-shadow: none;
+
+  .moj-filter__header {
+    background: none;
+    border: 1px solid white;
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/ui/components/filterGroup.scss:7]: `govuk-colour("white")`

File: app/frontend/src/lib/ui/components/filterGroup.scss [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  .moj-filter__heading-action .govuk-link {
+    font-size: 1rem;
+  }
+
+  .moj-filter__content {
+    .govuk-heading-s {
+      font-size: 1rem;
+      font-weight: bold;
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/ui/components/filterGroup.scss:21]: `$govuk-font-weight-bold` (or something like that)

File: app/frontend/src/lib/ui/components/filterGroup.scss [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+  .moj-filter-tags {
+    margin-top: govuk-spacing(1);
+  }
+
+  .moj-filter__tag:after {
+    height: 0;
+    width: 0;
+  }
+}
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/ui/components/filterGroup.scss:49]: newline

File: app/views/shared/_filter_group.html.haml [hunk 7]
```diff
@@ -1,76 +1,63 @@
 -# Usage to include filter group
 -#
--# = render partial: 'shared/filter_group', locals: {
+-# = render 'shared/filter_group', :f => <FormBuilder form>,
```
Comment (cpjmcquillan) [inline app/views/shared/_filter_group.html.haml:3]: Can be updated to modern style `key: value`

File: app/views/shared/_filter_group.html.haml [hunk 8]
```diff
@@ -1,76 +1,63 @@
...
-          .moj-filter__selected
+- if filters.present?
+  .moj-filter
+    .moj-filter__header
+      .selected-filters__heading
+        %span.moj-filter__heading-title
+          %h2.govuk-heading-s
+            = filters[:title]
+        %a.govuk-link.govuk-link--no-visited-state{ href: '', id: 'close-all-groups' }
+          = 'Close all'
```
Comment (cpjmcquillan) [inline app/views/shared/_filter_group.html.haml:21]: Can we I18n all of these please?

File: app/views/shared/_filter_group.html.haml [hunk 9]
```diff
@@ -1,76 +1,63 @@
...
+
+    - if filters[:totalCount] > 0 && filters[:removeButtons]
+      .moj-filter__content
+        .moj-filter__selected
+          .selected-filters__heading{ class: 'govuk-!-margin-bottom-2' }
+            .govuk-body{ class: 'govuk-!-margin-bottom-0' }
+              = 'Selected filters'
+            %a.govuk-link.govuk-link--no-visited-state{ href: '', id: 'clear-filters-button' }
```
Comment (cpjmcquillan) [inline app/views/shared/_filter_group.html.haml:29]: If this is a link to nowhere, would it be better to use a button styled as a link?

File: app/views/shared/_filter_group.html.haml [hunk 10]
```diff
@@ -1,76 +1,63 @@
...
+              = group[:title]
 
             %ul.moj-filter-tags
-              %li
-                %a.moj-filter__tag{ href: "/path/to/remove/this" }
-                  %span.govuk-visually-hidden
-                    = 'Remove this filter'
-                  = 'Art and design'
-                  %i.fa.fa-close
+              - group[:options].each do |school|
```
Comment (cpjmcquillan) [inline app/views/shared/_filter_group.html.haml:37]: If this is to be reused, I would change the variable name `school` to something that isn't so specific

File: spec/features/hiring_staff_can_filter_vacancies_in_their_dashboard_spec.rb [hunk 11]
```diff
@@ -48,7 +48,7 @@
       scenario 'it shows all published vacancies' do
         visit jobs_with_type_organisation_path(:published)
 
-        expect(page).to have_content('Showing all jobs')
```
Comment (cpjmcquillan) [inline spec/features/hiring_staff_can_filter_vacancies_in_their_dashboard_spec.rb:51]: Are we definitely getting rid of this text?

File: app/components/hiring_staff/vacancies_component.html.haml (hunk 3 above)
Comment (alexbowen) [reply app/components/hiring_staff/vacancies_component.html.haml:14]: it was removed as it is part of an old figma design

File: spec/features/hiring_staff_can_filter_vacancies_in_their_dashboard_spec.rb (hunk 11 above)
Comment (alexbowen) [reply spec/features/hiring_staff_can_filter_vacancies_in_their_dashboard_spec.rb:51]: again it was removed as it is part of an old figma design, i checked with chris

File: app/views/shared/_filter_group.html.haml (hunk 9 above)
Comment (alexbowen) [reply app/views/shared/_filter_group.html.haml:29]: yes good point, probably need to do a accessibility once over with all this 👍

File: app/components/hiring_staff/vacancies_component.html.haml (hunk 3 above)
Comment (cpjmcquillan) [reply app/components/hiring_staff/vacancies_component.html.haml:14]: Which Figma are you using?

File: app/components/hiring_staff/vacancies_component.html.haml [hunk 12]
```diff
@@ -2,22 +2,16 @@
   .govuk-grid-column-one-quarter
     = form_for @filters_form, url: organisation_managed_organisations_path, html: { method: "patch" } do |f|
       = render 'shared/filter_group', f: f,
-        filters: { title: 'Job filters', items: [{ title: 'Locations', attribute: :managed_school_ids, options: @school_options, value_method: :id, text_method: :name, small: true }] }
+        filters: { title: 'Filter results', :removeButtons => true, totalCount: @filters[:managed_school_ids]&.count, items: [{ title: 'Locations', key: 'locations', search: true, attribute: :managed_school_ids, selected: @filters[:managed_school_ids], options: @school_options, value_method: :id, text_method: :name, small: true }] }
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/vacancies_component.html.haml:5]: `removeButtons: true`

**Pull request merged**